docs/DeveloperGuide.adoc

```diff
@@ -295,35 +457,146 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |possess the ability to update existing food entries |update the existing food entries if there are changes in their prices/descriptions.
+
+|`* * *` |user |only be able to see what I can afford for meals and beverages |save time scrolling through meals that fit my budget.
+
+|`* *` |user |know the opening and closing timings of the food stores in NUS |closed shops are not recommended to me to prevent me from wasting time to travel to these shops.
+
+|`* *` |smart user |put aliases to the commands available in the application | personalise the app and use it more effectively.
+
+|`* *` |lazy user |obtain a recommended meal plan according to a specified daily allowance |save the time of having to plan for my meals for the day.
+
+|`* *` |user |add my savings of the month into a customised fund |purchase rewards/gifts/items that I require when I have saved enough for them.
+
+|`* *` |lazy user |load and save data from other computers |transfer data to an application onto another desktop.
+
+|`* *` |user with dietary/religious restrictions |exclude meals that do not fulfil my dietary requirements |reduce my options to only meals that I can consume.
+
+|`* *` |slow user |a guided tutorial to bring me through the basic functionality of the program |become more familiar with the program before I start using.
+
+|`* *` |Japanese food lover |prioritise Japanese food options over other similarly priced products |find specific food types of our choice/cravings we have.
+
+|`* *` |advanced user |shorten my commands |type faster and more efficiently.
+
+|`* *` |lazy user |have autocomplete |find food items without having to type long keywords.
+
+|`* *` |careful user |have a calendar function |keep track of the progress of my spending for the current month
+
+|`* *` |non-tech savvy user |have an easier way to understand how the works eg. through a video |use the app effectively without having to read long user guides.
+
+|`*`|dyslexic user |an app with easy to read font| use the app comfortably with being hindered by my reading disabilities.
+
+|`*` |user with ADHD |an app with simple commands and UI |use it comfortably without major distractions.
+
+|`*` |user who loves to customize things |have a theme changing function of the app from a list of themes available | personalise the app to a theme that I like.
+
+|`*` |user who hates travelling | sort places from the nearest to the furthest from my current location| find food places that are easy for me to get to.
+
+|`*` |user |add the birthdays of my friends |set up reminders to buy gifts for my friends.
+
+|`*` |user |receive sample suggestions and examples to understand how to use the program. |understand how to user the application effectively.
+
+|`*` |forgetful user |save specific meal sets to reuse |save time on inputting my meals daily.
+|=======================================================================
 
```

Comment:
Perhaps you could rearrange the user in alphabetical order to aid in understanding what a role can do.